sdks/java/io/solr/src/test/java/org/apache/beam/sdk/io/solr/SolrIOIT.java

```diff
@@ -0,0 +1,178 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.solr;
+
+import static org.apache.beam.sdk.testing.SourceTestUtils.readFromSource;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+
+import org.apache.beam.sdk.io.BoundedSource;
+import org.apache.beam.sdk.io.common.IOTestPipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.SourceTestUtils;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.request.CollectionAdminRequest;
+import org.apache.solr.client.solrj.response.CollectionAdminResponse;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrInputDocument;
+import org.joda.time.Instant;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A test of {@link SolrIO} on an independent Solr instance.
+ *
+ * <p>This test requires a running instance of Solr (./bin/solr start -e cloud -noprompt)
+ *
+ * <p>You can run this test by doing the following from the beam parent module directory:
+ *
+ * <pre>
+ *  mvn -e -Pio-it verify -pl sdks/java/io/solr -DintegrationTestPipelineOptions='[
+ *  "--solrZookeeperServer=127.0.0.1:9983"]'
+ * </pre>
+ */
+public class SolrIOIT {
+
+  public static final String READ_COLLECTION = "beam";
+  public static final int NUM_SHARDS = 3;
+  public static final long NUM_DOCS = 60000;
+  public static final String WRITE_COLLECTION = "beam" + Instant.now().getMillis();
+  private static final Logger LOG = LoggerFactory.getLogger(SolrIOIT.class);
+
+  private static AuthorizedSolrClient solrClient;
+  private static IOTestPipelineOptions options;
+  private static SolrIO.ConnectionConfiguration readConnectionConfiguration;
+  @Rule public TestPipeline pipeline = TestPipeline.create();
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    PipelineOptionsFactory.register(IOTestPipelineOptions.class);
+    options = TestPipeline.testingPipelineOptions().as(IOTestPipelineOptions.class);
+    readConnectionConfiguration = SolrIO.ConnectionConfiguration
+        .create(options.getSolrZookeeperServer(), READ_COLLECTION);
```

Comment:
Yeah, I think this will be appropriate to move collection name to Read and Write.